File: exercism/typescript/bob/bob.ts
const QUESTION_MARK: string = "?";
const ANY_UPPERCASE_LETTER = /[A-Z]/g;

function isQuestion(msg: string): boolean {
  return msg.trim().endsWith(QUESTION_MARK);
}

function isYelling(msg: string): boolean {
  return msg.toUpperCase() === msg && ANY_UPPERCASE_LETTER.test(msg);
}

function isNothingSpecific(msg: string): boolean {
  return msg.trim().length === 0;
}

export function hey(message: string): string {

  if (isNothingSpecific(message)) {
    return 'Fine. Be that way!';
  }

  if (isYelling(message)) {
    return isQuestion(message) ? 'Calm down, I know what I\'m doing!' : 'Whoa, chill out!'; 
  }

  return isQuestion(message) ? 'Sure.' : 'Whatever.';
}

File: exercism/python/bob/bob.py
def response(hey_bob: str):
    is_question = hey_bob.strip().endswith('?')

    if len(hey_bob.strip()) == 0:
        return 'Fine. Be that way!'
    elif hey_bob.upper() == hey_bob and any(char.isalpha() for char in hey_bob):
        return 'Calm down, I know what I\'m doing!' if is_question else 'Whoa, chill out!'
    else:
        return 'Sure.' if is_question else 'Whatever.'
